File: Sources/CxxLib/Calculator.cpp
#include "include/Calculator.hpp"

Calculator::Calculator() = default;

int Calculator::add(int a, int b) {
    return a + b;
}

int Calculator::subtract(int a, int b) {
    return a - b;
}

double Calculator::multiply(double a, double b) {
    return a * b;
}

double Calculator::divide(double a, double b) {
    if (b == 0) {
        return 0.0; // Simple error handling
    }
    return a / b;
} 

File: Sources/CxxSPM/Calculator.swift
import CxxLib

public class SwiftCalculator {
    private var calculator: Calculator
    
    public init() {
        calculator = Calculator()
    }
    
    public func add(_ a: Int32, _ b: Int32) -> Int32 {
        return calculator.add(a, b)
    }
    
    public func subtract(_ a: Int32, _ b: Int32) -> Int32 {
        return calculator.subtract(a, b)
    }
    
    public func multiply(_ a: Double, _ b: Double) -> Double {
        return calculator.multiply(a, b)
    }
    
    public func divide(_ a: Double, _ b: Double) -> Double {
        return calculator.divide(a, b)
    }
} 